service/mapper/transitions.go

```diff
@@ -58,19 +59,34 @@ func NewTransitions(log zerolog.Logger, load Loader, chain dps.Chain, feed Feede
 		load:  load,
 		chain: chain,
 		feed:  feed,
-		index: index,
+		read:  read,
+		write: write,
 		once:  &sync.Once{},
 	}
 
 	return &t
 }
 
-// BootstrapState bootstraps the state by loading the checkpoint if there is one and initializing
-// the elements subsequently used by the FSM.
-func (t *Transitions) BootstrapState(s *State) error {
+// InitializeMapper will initialize the mapper by either going into
+// bootstrapping or into resuming, depending on the configuration.
+func (t *Transitions) InitializeMapper(s *State) error {
+	if s.status != StatusInitialize {
+		return fmt.Errorf("invalid status for initializing mapper (%s)", s.status)
+	}
+
+	if t.cfg.BootstrapState {
+		s.status = StatusBootstrap
+		return nil
+	}
 
-	// Bootstrapping should only happen when the state is empty.
-	if s.status != StatusEmpty {
+	s.status = StatusResume
+	return nil
+}
```

Comment:
I'm confused with when it is that you start a func with a newline and when it is that you do not 🤔 It seems inconsistent to me. I've always seen functions not start with a newline, for me the fact that it's at a different tabulation level is enough to separate it from the function prototype line, so I never start any function with a newline personally.

Reply:
I generally always start a function with a new line, unless the first line(s) are part of the "pattern" for that function type. In this case, they all check whether the status is correct, before doing anything else, so I like to get that out of the way before starting the "real" function body.